### ISSUE TITLE
CocoaPods could not find compatible versions for pod "GTMSessionFetcher/Core"

### ISSUE THREAD

**Issue by atif-infinityup** [googlesamples/google-signin-unity#225]
I am using Google SignIn Plugin version 1.0.4 Upgraded and Firebase Authentication (Version: 10.3.0) for Google SignIn on IOS using unity version 2021.3.16f1. 
Unity is unable to make Xcode project properly. 
Showing Error
"iOS framework addition failed due to a CocoaPods installation failure. This will will likely result in a non-functional Xcode project."
<img width="616" alt="Screenshot 2023-01-18 at 11 25 37 PM" src="https://user-images.githubusercontent.com/123011790/213263743-db94c3e2-d63c-4f1f-8af1-453f2a166a83.png">

Then I tried to install pod from terminal the issue remains the same.

<img width="569" alt="Screenshot 2023-01-18 at 11 27 09 PM" src="https://user-images.githubusercontent.com/123011790/213264039-44303c3a-673b-4a19-858c-698a2ae838c1.png">

So basically pod is unable to find correct GTMSessionFetcher/Core file due to conflict between two requirements. 
Is there any way to increase GTMSessionFetcher/Core (~> 1.1) value ?

**Comment by ChrisBindy** [comment]
Morning I just had this same problem. I have managed to work around this by manually modifying the locally install PodSpecs to use a newer version of GTMSessionFetcher/Core that matches FirebaseAuth.

Find the location of the GoogleSignIn podspec file. Mine was at "/Users/{user}/.cocoapods/repos/cocoapods/Specs/d/4/0/GoogleSignIn/5.0.2/GoogleSignIn.podspec.json"

and change `"GTMSessionFetcher/Core": "~> 1.1"` under dependencies to `"GTMSessionFetcher/Core": [
      ">= 2.1",
      "< 4.0"
    ]`

this will make it match the FirebaseAuth podspec and the xcode workspace will be correctly created .

**Comment by cinephile231** [comment]
After chaing firebase sdk version 9.6.0, error is gone.

**Comment by xmx77** [comment]
Any one fix this issue ? 
@atif-infinityup  can you tell me how you fix this issue ?


I can't understand why this plugin not update since 2018 !!

**Comment by s-saens** [comment]
## Resolve
I found we can solve this issue only by downgrading Firebase to version 9.6.0. And we don't have to reinstall Firebase packages. Just change **pod versions** in `*Dependencies.xml` file to 9.6.0, like this:

<img width="896" alt="image" src="https://user-images.githubusercontent.com/50483226/225494771-925194e0-087a-46f8-b68c-820339f86c75.png">

---

## Why we can resolve this issue only by downgrading Firebase:

We can find pod dependencies on [this site](https://cocoapods.org/), and I found some issues.

First, "FirebaseAuth" pod is depends on "GTMSessionFetcher/Core" with version is: "2.1 <= version < 4.0".
Second, in this google-signin-unity package, it uses "GoogleSignIn" pod. In the change log, I could find this:
<img width="709" alt="image" src="https://user-images.githubusercontent.com/50483226/225489073-062621d8-e4c1-4f85-9a62-c413e3cc536d.png">
this means when this pod version is under 6.2.4, it can only depend on "GTMSessionFetcher/Core" with version 1.x.

So I tried to build with GoogleSignIn version 6.2.4, but this error occurred.
```
Libraries/Plugins/iOS/GoogleSignIn/GoogleSignIn.h:18:17: error: no type or protocol named 'GIDSignInDelegate'
```
and in the change log of "GoogleSignIn" pod, I could find this
<img width="901" alt="image" src="https://user-images.githubusercontent.com/50483226/225489790-8b991783-67fe-4448-8145-f59bc32694ee.png">

So I cannot use the version 6.2.4.
But we cannot use the version under 6.2.4 because of "GTMSessionFetcher/Core" version conflict with FirebaseAuth.
And I don't think this plugin will be updated...

So the only remain option is downgrading Firebase to 9.6.0.

As we can see [here](https://firebase.google.com/support/release-notes/ios#version_1000_-_october_10_2022), On Firebase version 10.0.0 it changed to use GTMSessionFetcher only with version over 2.x. This means we can solve this error if we use Firebase version under 10.0.0.

**Comment by ubberkid** [comment]
Although downgrading firebase to 9.6.0 will fix it.. 

I would definitely just do the fix @ChrisBindy mentioned.
I think staying up-to-date is more important as there are many bug fixes in latest versions

**Comment by yusufisik** [comment]
I have found a solution:
https://github.com/googlesamples/google-signin-unity/issues/228#issuecomment-1593791767

**Comment by N4TUR1NO** [comment]
> and change `"GTMSessionFetcher/Core": "~> 1.1"` under dependencies to `"GTMSessionFetcher/Core": [ ">= 2.1", "< 4.0" ]`

[ChrisBindy](https://github.com/ChrisBindy) is any other actions required? i changed dependencies like you said and nothing happens ;(

**Comment by ChrisBindy** [comment]
@N4TUR1NO Heya, unfortunately I don't work with Unity anymore so I'm unsure of the fix, I hope you mange to get it sorted

**Comment by varLG** [comment]
> Morning I just had this same problem. I have managed to work around this by manually modifying the locally install PodSpecs to use a newer version of GTMSessionFetcher/Core that matches FirebaseAuth.
> 
> Find the location of the GoogleSignIn podspec file. Mine was at "/Users/{user}/.cocoapods/repos/cocoapods/Specs/d/4/0/GoogleSignIn/5.0.2/GoogleSignIn.podspec.json"
> 
> and change `"GTMSessionFetcher/Core": "~> 1.1"` under dependencies to `"GTMSessionFetcher/Core": [ ">= 2.1", "< 4.0" ]`
> 
> this will make it match the FirebaseAuth podspec and the xcode workspace will be correctly created .



I found same folder in trunk (from: cocoapods/repos/trunk/Specs/d/4/0/GoogleSignIn/6.0.2(my version)) and worked for my case. don't forget the edit main file like upper comment.

my pod file is:
 

> source 'https://cdn.cocoapods.org/'
> 
>platform :ios, '13.0'
> 
>target 'UnityFramework' do
>pod 'FBSDKCoreKit', '~> 16.0.1'
>pod 'FBSDKCoreKit_Basics', '~> 16.0.1'
>pod 'FBSDKGamingServicesKit', '~> 16.0.1'
>pod 'FBSDKLoginKit', '~> 16.0.1'
>pod 'FBSDKShareKit', '~> 16.0.1'
>pod 'Firebase/Analytics', '10.25.0'
>pod 'Firebase/Auth', '10.25.0'
>pod 'Firebase/Core', '10.25.0'
>pod 'Firebase/Messaging', '10.25.0'
>pod 'GoogleSignIn', '~> 6.0.2'
>end
>target 'Unity-iPhone' do
>end
>use_frameworks! :linkage => :static


my googlesignin patch is
[https://github.com/CodeMasterYi/google-signin-unity/releases](url)

**Comment by 32nguyen** [comment]
@All: Thanks for all. @varLG what do you mean by
> comment "don't forget the edit main file like upper " 

Did you cross them out?
